### PR TITLE
Tr2 15

### PR DESCRIPTION
**Pull request by aczender**

### CODE REVIEW

**Review by starlinginferno — APPROVED** [review]
would prefer separate tables for the connection but good job!

File: src/main/java/rueppellii/backend2/tribes/resource/ResourceService.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    @Autowired
+    public ResourceService(ResourceRepository resourceRepository) {
+        this.resourceRepository = resourceRepository;
+    }
+
+    public Resource getResourceById(Long id) {
+        Optional<Resource> resource = resourceRepository.findById(id);
+        if (resource.isPresent()) {
+            return resource.get();
+        } else throw new IllegalArgumentException();
```
Comment (csadesz) [inline src/main/java/rueppellii/backend2/tribes/resource/ResourceService.java]: else unneccessary

File: src/main/java/rueppellii/backend2/tribes/resource/ResourceService.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            resourceRepository.save(resource);
+            return ResponseEntity.status(HttpStatus.OK).build();
+        }
+        return ResponseEntity.status(HttpStatus.BAD_REQUEST).build();
+    }
+
+    public boolean validateType(Resource resource) {
+        return resource.getResource_type() == ResourceType.RESOURCE_FOOD || resource.getResource_type() == ResourceType.RESOURCE_GOLD;
+    }
+
```
Comment (csadesz) [inline src/main/java/rueppellii/backend2/tribes/resource/ResourceService.java:43]: empty lines

File: src/test/java/rueppellii/backend2/tribes/resource/ResourceServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(SpringExtension.class)
+public class ResourceServiceTest {
+
+    private ResourceService resourceService;
+    private Resource resource;
+    private ResponseEntity r;
```
Comment (csadesz) [inline src/test/java/rueppellii/backend2/tribes/resource/ResourceServiceTest.java]: I think in some case "response" would be more readable than "r"

for example:

.... .isEqualTo(response);  (like a book 😄)

File: src/main/java/rueppellii/backend2/tribes/resource/Resource.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    @Enumerated(EnumType.STRING)
+    private ResourceType resource_type;
+
+    @NotBlank
+    private Integer amount;
+
+    private Timestamp updated_at;
+
+    @ManyToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "kingdom_id", nullable = false)
```
Comment (csadesz) [inline src/main/java/rueppellii/backend2/tribes/resource/Resource.java]: prefer separate tables for the connection as Lili said :)

**Review by csadesz — CHANGES_REQUESTED** [review]

**Review by csadesz — APPROVED** [review]

**Pull request merged**